### PR TITLE
Add `use_posterior_mean` option to scanvi `predict`

### LINKED ISSUE #1587: Stochastic/Non-deterministic prediction for scANVI
**Is your feature request related to a problem? Please describe.**
I am trying to run a procedure called self projection with a scANVI model. The procedure entails taking a trained scANVI model and repeatedly generating predictions on the dataset used to train it, hence the name 'self projection.' Because scANVI employs a probabilistic framework and not a deterministic one, I would think that each time scANVI generates predictions on some dataset (i.e. scANVI.predict()), the predictions would be slightly different each time. However, when I repeatedly use scANVI.predict(), it generates the same predictions each time.

**Describe the solution you'd like**
I looked through the documentation for scANVI, but I do not see any option/parameter for .predict() function that allows this feature. I tried changing the seed but no change. Having such an option would be extremely helpful! Please let me know if there is a current work-around.

Thank you.

### PR DESCRIPTION
**Pull request by martinkim0**
-   [x] Closes #1587 (Replace xxxx with the GitHub issue number)
-   [x] Tests added and passed if fixing a bug or adding a new feature
-   [x] All code checks passed.
-   [x] Added type annotations to new arguments/methods/functions.
-   [x] Added an entry in the latest `docs/release_notes/index.md` file if fixing a bug or adding a new feature.
-   [x] If the changes are patches for a version, I have added the `on-merge: backport to x.x.x` label.

### Summary by CodeRabbit

**New Feature**
- Added a new argument `use_posterior_mean` to the `predict` method in the `SCANVI` class. This allows for stochastic prediction of cell type labels.

**Bug Fix**
- Replaced `torch.log` with `torch.log1p` in `_scanvae.py` to avoid numerical instability.

**Test**
- Introduced a new test function `test_scanvi_predict_use_posterior_mean()` to validate the functionality of the new feature.

> 🎉 Here's to the code that's ever so lean,  
> With `use_posterior_mean`, predictions are keen.  
> No more instability, thanks to `log1p`,  
> Our tests ensure we don't miss a blip! 🚀

### CODE REVIEW

**Comment by cane11** [comment]
The code change makes sense and I was also puzzled by this decision in the past. However, are there any tasks where the non-default value has shown added value?
